src/gentropy/dataset/study_locus.py

```diff
@@ -169,13 +178,30 @@ def validate_study(self: StudyLocus, study_index: StudyIndex) -> StudyLocus:
                 self.df.join(
                     study_flags, f.col("studyId") == f.col("study_studyId"), "left"
                 )
-                # Flagging loci with failed studies:
+                # Flagging loci with flagged studies - without propagating the actual flags:
                 .withColumn(
                     "qualityControls",
                     StudyLocus.update_quality_flag(
                         f.col("qualityControls"),
                         f.size(f.col("study_qualityControls")) > 0,
-                        StudyLocusQualityCheck.FAILED_STUDY,
+                        StudyLocusQualityCheck.FLAGGED_STUDY,
+                    ),
+                )
+                # Flagging top-hits, where the study has available summary statistics:
+                .withColumn(
+                    "qualityControls",
+                    StudyLocus.update_quality_flag(
+                        f.col("qualityControls"),
+                        # Condition is true, if the study has summary statistics available and the locus is a top hit:
+                        f.array_contains(
+                            f.col("qualityControls"),
+                            StudyLocusQualityCheck.TOP_HIT.value,
+                        )
+                        & f.array_contains(
+                            f.col("study_qualityControls"),
+                            StudyQualityCheck.SUMSTATS_NOT_AVAILABLE.value,
```

Comment:
Does the logic holds here?

`TOP_HIT` & `SUMSTATS_NOT_AVAILABLE` => `SUMMARY_STATISTICS_AVAILABLE` 

MIssing `~` perhaps?